### PR TITLE
Upgrade argonaut dependency

### PR DESCRIPTION
**Pull request by thomashoneyman**
The most recent [major release of the Argonaut library](https://github.com/purescript-contrib/purescript-argonaut-codecs/releases/tag/v7.0.0) introduces typed errors. This PR updates this library for compatibility.

This library will need a new release to stay compatible with the next package set, as tracked by this issue:
https://github.com/purescript/package-sets/issues/642

Sorry for the hassle! Thanks!

### CODE REVIEW

File: src/PscIde/Command.purs [hunk 1]
```diff
@@ -283,8 +283,8 @@ data ImportType = Implicit | Explicit (Array String) | Hiding (Array String)
 data ImportResult = SuccessFile Message | SuccessText (Array String) | MultipleResults (Array TypeInfo)
 
 unwrapResponse :: forall a b. DecodeJson a => DecodeJson b => String -> Either String (Either a b)
-unwrapResponse s = do
-  json <- jsonParser s
+unwrapResponse s = lmap printJsonDecodeError do
```
Comment (thomashoneyman) [inline src/PscIde/Command.purs:286]: I opted to recover the string here as this is used in the later `makeAff` calls. To minimize changes I thought this was the appropriate place to return from `JsonDecodeError` to the previous `String` type.

**Comment by thomashoneyman** [comment]
The test failure is because I didn't update the psc-package file. However, when the next package set comes out (in the next week or so) the psc-package file could be updated to point at that set instead. Let me know how you'd like to handle this and I can do my part to help out!

**Comment by kritzcreek** [comment]
> Sorry for the hassle

No worries, giving argonaut typed errors is a great change!

> Let me know how you'd like to handle this

Whichever is less of a hassle for you. Eventually we'll want to move this library to `spago` (I think the editor plugins for vscode and the language server have already done so), but just keeping CI running with psc-package after the next packaget-set is good enough for me. Let's just make sure to open an issue on this repo so I don't forget.

File: packages.dhall [hunk 2]
```diff
@@ -0,0 +1,12 @@
+let upstream =
+      https://github.com/purescript/package-sets/releases/download/psc-0.13.8-20200615/packages.dhall sha256:5d0cfad9408c84db0a3fdcea2d708f9ed8f64297e164dc57a7cf6328706df93a
+
+let overrides = 
+      { argonaut = upstream.argonaut // { version = "v7.0.0" }
+      , argonaut-codecs = upstream.argonaut-codecs // { version = "v7.0.0" }
+      , argonaut-traversals = upstream.argonaut-traversals // { version = "v8.0.0" }
+      }
```
Comment (thomashoneyman) [inline packages.dhall]: These overrides can be removed once the next package set is released -- you'd run `spago upgrade-set` and delete these lines.

File: spago.dhall [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{ name = "psc-ide"
+, dependencies =
+  [ "aff"
+  , "arrays"
+  , "console"
+  , "effect"
```
Comment (thomashoneyman) [inline spago.dhall:7]: All the other dependencies were already specified in the Bower file, but this one is new. Spago added it when generating the config as it was included transitively, so I left it in place.

File: README.md [hunk 4]
```diff
@@ -1,16 +1,12 @@
-[![Build Status](https://travis-ci.org/kRITZCREEK/purescript-psc-ide.svg?branch=master)](https://travis-ci.org/kRITZCREEK/purescript-psc-ide)
-
-purescript-psc-ide
-===
+# purescript-psc-ide [![Build Status](https://travis-ci.org/kRITZCREEK/purescript-psc-ide.svg?branch=master)](https://travis-ci.org/kRITZCREEK/purescript-psc-ide)
```
Comment (thomashoneyman) [inline README.md:1]: Updated the Markdown file according to markdownlint's rules, but I can revert these changes if you'd like. Just thought I'd do it while I'm already linting things.

**Comment by thomashoneyman** [comment]
OK, this is super weird -- it's failing to build the argonaut-codecs dependency because generics-rep is missing. But argonaut-codecs _depends_ on generics-rep and builds fine in all the other projects I've been testing. I'm not sure what's going on. It also builds fine if I install generics-rep in this library, but that shouldn't be necessary.

**Comment by kritzcreek** [comment]
Sorry I'm kind of swamped at work, so don't have the time to help debug this right now.

**Comment by thomashoneyman** [comment]
I was wrong and this was an introduced dependency after all. I've indicated that in the overrides -- when the next package set is released the overrides can be removed.

I'm not sure why the CI build status isn't showing up on GitHub, but you can see that the build does pass now:
https://travis-ci.org/github/kritzcreek/purescript-psc-ide

**Comment by kritzcreek** [comment]
Thanks, looking good.

**Comment by kritzcreek** [comment]
Released as v16.0.0

**Pull request merged**